cmd/cstor-pool-mgmt/controller/pool-controller/after_queue_handler.go

```diff
@@ -0,0 +1,192 @@
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+)
+
+// syncHandler compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the cStorPoolUpdated resource
+// with the current status of the resource.
+func (c *CStorPoolController) syncHandler(key, operation string) error {
+	glog.Infof("at sync handler")
+	cStorPoolGot, err := c.getPoolResource(key)
+	if err != nil {
+		return err
+	}
+	status, err := c.cStorPoolEventHandler(operation, cStorPoolGot)
+	if status == common.StatusIgnore {
+		return nil
+	}
+	cStorPoolGot.Status.Phase = status
+	if err != nil {
+		_, err := c.clientset.OpenebsV1alpha1().CStorPools().Update(cStorPoolGot)
+		if err != nil {
+			return err
+		}
+		return err
+	}
+	_, err = c.clientset.OpenebsV1alpha1().CStorPools().Update(cStorPoolGot)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// cStorPoolEventHandler is to handle cstor pool related events.
+func (c *CStorPoolController) cStorPoolEventHandler(operation string, cStorPoolGot *apis.CStorPool) (string, error) {
+	pool.RunnerVar = util.RealRunner{}
+	switch operation {
+	case "add":
+		glog.Info("added event")
+		// CheckValidPool is to check if pool attributes are correct.
+		err := pool.CheckValidPool(cStorPoolGot)
+		if err != nil {
+			return common.StatusOffline, err
+		}
+
+		// ImportPool is to try importing pool.
+		err = pool.ImportPool(cStorPoolGot)
+		if err == nil {
+			glog.Infof("Import Pool successful")
+			return common.StatusOnline, nil
+		}
+
+		// IsInitStatus is to check if initial status of cstorpool object is `init`.
+		if IsInitStatus(cStorPoolGot) {
+			// LabelClear is to clear pool label
+			err = pool.LabelClear(cStorPoolGot.Spec.Disks.DiskList)
+			if err != nil {
+				glog.Infof("Unable to clear pool labels : %v", err.Error())
+			}
+
+			// CreatePool is to create cstor pool.
+			err = pool.CreatePool(cStorPoolGot)
+			if err != nil {
+				return common.StatusOffline, err
+			}
+			glog.Infof("Pool creation successful")
+			return common.StatusOnline, nil
+		}
+		break
+
+	case "modify":
+		glog.Info("modify event") //ignoring as of now
+		break
+
+	case "destroy":
+		glog.Info("destroy event")
+		// DeletePool is to delete cstor pool.
+		err := pool.DeletePool("cstor-" + string(cStorPoolGot.ObjectMeta.UID))
+		if err != nil {
+			return common.StatusDeletionFailed, err
+		}
+		// removeFinalizer is to remove finalizer of cStorPool resource.
+		err = c.removeFinalizer(cStorPoolGot)
+		if err != nil {
+			return common.StatusOffline, err
+		}
+		return common.StatusIgnore, nil
+	}
+
+	return common.StatusIgnore, nil
+}
+
+// enqueueCstorPool takes a CStorPool resource and converts it into a namespace/name
+// string which is then put onto the work queue. This method should *not* be
+// passed resources of any type other than CStorPools.
+func (c *CStorPoolController) enqueueCStorPool(obj interface{}, q common.QueueLoad) {
```

Comment:
This method should ideally be in new_pool_controller.go file.

Reply:
will look into